ovn_cluster.sh

```diff
@@ -716,34 +716,34 @@ function build-images-with-ovn-rpms() {
 }
 
 function build-images-with-ovn-sources() {
-    if [ ! -d ./ovs ]; then
-	    echo "OVS_SRC_PATH = $OVS_SRC_PATH"
-	    if [ "${OVS_SRC_PATH}" = "" ]; then
-            echo "Set the OVS_SRC_PATH var pointing to the location of ovs source code."
-            exit 1
-	    fi
-
-	    rm -rf ./ovs
-	    cp -rf $OVS_SRC_PATH ./ovs
-	    DO_RM_OVS='yes'
+    echo "OVS_SRC_PATH = $OVS_SRC_PATH"
+    if [ ! -d "${OVS_SRC_PATH}" ]; then
+        echo "Set the OVS_SRC_PATH var pointing to the location of ovs source code."
+        exit 1
+    fi
+
+    echo "OVN_SRC_PATH = $OVN_SRC_PATH"
+    if [ ! -d "${OVN_SRC_PATH}" ]; then
+        echo "Set the OVN_SRC_PATH var pointing to the location of ovn source code."
+        exit 1
     fi
 
-    if [ ! -d ./ovn ]; then
-	    echo "OVN_SRC_PATH = $OVN_SRC_PATH"
-	    if [ "${OVN_SRC_PATH}" = "" ]; then
-            echo "Set the OVN_SRC_PATH var pointing to the location of ovn source code."
-            exit 1
-	    fi
-	    rm -rf ovn
-	    cp -rf $OVN_SRC_PATH ovn
-	    DO_RM_OVN='yes'
+    if [ -d ./ovs ]; then
```

Comment:
rm -rf can be safely done if the file is there, let's remove this conditional

Reply:
> Interesting request. I'm okay with it, but it changes a little bit the existing behavior on
> explicitly re-using the src/ovn src/ovs directories if they were present. I'm guessing you don't want that, right?
> 
> Do you see a situation when that could be wanted and if so, have a new knob added that would honor the existing src directories?
> 
> Another way to approach this is to simply fail if the destination directories are present and force the user to remove them and retry. That would be safer but potentially more annoying. wdyt?

The use case that I have is that I have made a change to OVN code and I was expecting to rebuild based on this new code. However, unexpectedly - to me anyway, it was not rebuilding.

I think the fail would be more annoying but that is just for my use case.

I suppose the disadvantage of recreating the directories each time is performance?

Reply:
> > Interesting request. I'm okay with it, but it changes a little bit the existing behavior on
> > explicitly re-using the src/ovn src/ovs directories if they were present. I'm guessing you don't want that, right?
> > Do you see a situation when that could be wanted and if so, have a new knob added that would honor the existing src directories?
> > Another way to approach this is to simply fail if the destination directories are present and force the user to remove them and retry. That would be safer but potentially more annoying. wdyt?
> 
> The use case that I have is that I have made a change to OVN code and I was expecting to rebuild based on this new code. However, unexpectedly - to me anyway, it was not rebuilding.
> 
> I think the fail would be more annoying but that is just for my use case.
> 
> I suppose the disadvantage of recreating the directories each time is performance?

nah, not an issue. I'm okay with merging the changes after making the changes we talked about.

Reply:
I reworked this as a compromise - should cover both use cases.